tendrl/commons/objects/__init__.py

```diff
@@ -83,11 +83,13 @@ def load_definition(self):
 
     @thread_safe
     def save(self, update=True, ttl=None):
+        hash_key_changed = True
         if "Message" not in self.__class__.__name__:
             # If local object.hash is equal to
             # central_store object.hash, return
             if self.hash_compare_with_central_store(ttl=ttl):
-                return
```

Comment:
If there is no change in hash keys then we are actually not updating object. But we have to update the watcher fields. Because if any watcher filed have TTL like node_context then it will be deleted when there is no change in an object. Because object like node_context fields won't change frequently. When components like node-agent or gluster-integration start if we do load and save objects then no changes in old objects fields and local objects then watcher attributes won't be created anymore, It will change when any one field in object is changed. So in this new logic object won't be updated but watcher fields are updated and if there is no watcher thread then it will create new one.

Reply:
objects like node_context, bricks and volumes we are actually doing load when node-agent or gluster-integration is restarted.